### PR TITLE
refactor: renamed classes and removed redundant options

### PR DESCRIPTION
**Pull request by StevenMaio**
CollisionCounter renamed to Participant. Removed code that generated
circuits.

### CODE REVIEW

File: pairing.py [hunk 1]
```diff
@@ -5,64 +5,48 @@
 
 from random import choices
 
-class CollisionCounter:
+class Participant:
 
     def __init__(self, num_people, gamma=1/np.e):
```
Comment (qwinters) [inline pairing.py:10]: If this class = 1 individual participant, then I'm not too big of a fan of having each person have their own `num_people` and `gamma` parameters. To me it's a little confusing + implies that we could have separate gamma parameters for separate people. 

Maybe we just need a name that's something other than `Participant`

File: pairing.py [hunk 2]
```diff
@@ -5,64 +5,48 @@
...
-            meeting_counts[i].increment_hits(j)
-            meeting_counts[j].increment_hits(i)
-    print("average hit/person:",
-            (2 if circuit else 1)*num_iters/(num_people-1))
-    for i,q in enumerate(meeting_counts):
-        cc = q.collision_counter()
-        print("person {}:".format(i), cc)
+            participants[i].increment_hits(j)
+            participants[j].increment_hits(i)
+    print("average hit/person:", num_iters/(num_people-1))
```
Comment (qwinters) [inline pairing.py:46]: nit: if we're changing this up, can we switch to using a proper logger?

File: pairing.py (hunk 1 above)
Comment (StevenMaio) [reply pairing.py:10]: Oh, yeah. I wasn't even thinking about `num_people`, but I realized that each person having their own `gamma` is redundant.

File: pairing.py [hunk 3]
```diff
@@ -5,64 +5,48 @@
...
+    if len(unmatched)==1:
+        pairs[0].append(unmatched[0])
     return pairs
 
 # sampling first person based on stuff
-def main(num_people, num_iters, gamma, circuit):
-    meeting_counts = [CollisionCounter(num_people, gamma=gamma) for k in range(num_people)]
+def main(num_people, num_iters, gamma):
+    participants = [Participant(num_people, gamma=gamma) for k in range(num_people)]
     for i in range(num_iters):
```
Comment (qwinters) [inline pairing.py:41]: nit: descriptive variables names please! 

Also i is reused later on in the for loop (which can be fixed by using descriptive variable names :) )

File: pairing.py [hunk 4]
```diff
@@ -5,64 +5,48 @@
...
-        if circuit:
-            pairs = generate_circuit(meeting_counts, num_people)
-        else:
-            pairs = generate_pairs(meeting_counts, num_people)
+        pairs = generate_pairs(participants)
         for i,j in pairs:
```
Comment (qwinters) [inline pairing.py:43]: nit: descriptive variable names please!

File: pairing.py [hunk 5]
```diff
@@ -5,64 +5,48 @@
 
 from random import choices
```
Comment (qwinters) [inline pairing.py:6]: nit: why do we have choices from random and `np.random.choice(...)`

Can we just use numpy's random class?

**Review by qwinters — CHANGES_REQUESTED** [review]
Can we also have add a test in this first real commit?

Just a super simple unit test -- I think the main method won't work right now (at time of this review) because it calls for a variable `p` that doesn't exist in the scope of the `main()` function

File: pairing.py (hunk 5 above)
Comment (StevenMaio) [reply pairing.py:6]: `np.random.choice` only samples uniformly

File: pairing.py (hunk 1 above)
Comment (qwinters) [reply pairing.py:10]: I think a good quick fix for now is to just use a universal constant btw -- the `networkx` stuff and more indepth changes we can address later

File: pairing.py (hunk 5 above)
Comment (qwinters) [reply pairing.py:6]: it has a weighted version: https://numpy.org/doc/stable/reference/random/generated/numpy.random.choice.html

the argument is `p`

File: pairing.py [hunk 6]
```diff
@@ -5,64 +5,48 @@
...
-    print("average hit/person:",
-            (2 if circuit else 1)*num_iters/(num_people-1))
-    for i,q in enumerate(meeting_counts):
-        cc = q.collision_counter()
-        print("person {}:".format(i), cc)
+            participants[i].increment_hits(j)
+            participants[j].increment_hits(i)
+    print("average hit/person:", num_iters/(num_people-1))
+    for i,q in enumerate(participants):
+        p = q.meetings_counter
```
[the remark this answers is not in the document]
Comment (StevenMaio) [reply pairing.py:48]: `p` is defined right there?

File: pairing.py [hunk 7]
```diff
@@ -3,66 +3,48 @@
...
-        for i,j in pairs:
-            meeting_counts[i].increment_hits(j)
-            meeting_counts[j].increment_hits(i)
-    print("average hit/person:",
-            (2 if circuit else 1)*num_iters/(num_people-1))
-    for i,q in enumerate(meeting_counts):
-        cc = q.collision_counter()
-        print("person {}:".format(i), cc)
+def main(num_people, num_iters, gamma):
+    gamma = np.log(gamma)
```
Comment (qwinters) [inline pairing.py:39]: nit: if we're doing this here can we change the `gamma` parameter in the method `main(...)` to something else? I feel like this line is super confusing without context

**Review by qwinters — APPROVED** [review]
👍 LGTM

**Pull request merged**